### PR TITLE
state: fix freed VIP table id index

### PR DESCRIPTION
**Pull request by kyhavlov**
This PR fixes a bug that could happen when the most recently assigned virtual IP was freed and reassigned.

### CODE REVIEW

**Review by thisisnotashwin — APPROVED** [review]

File: agent/consul/state/catalog_test.go [hunk 1]
```diff
@@ -1684,6 +1684,143 @@ func TestStateStore_EnsureService_virtualIps(t *testing.T) {
...
+	// Retrieve and verify
+	_, out, err = s.NodeServices(nil, "node1", nil)
+	assert.Nil(err)
+	assert.NotNil(out)
+
+	taggedAddress = out.Services["redis"].TaggedAddresses[structs.TaggedAddressVirtualIP]
+	assert.Equal(vip, taggedAddress.Address)
+	assert.Equal(ns2.Port, taggedAddress.Port)
+
+	// Delete the first service and make sure it no longer has a virtual IP assigned.
```
Comment (freddygv) [inline agent/consul/state/catalog_test.go:1758]:
```suggestion
	// Delete the last service and make sure it no longer has a virtual IP assigned.
```

**Review by freddygv — APPROVED** [review]
LGTM, left one minor comment

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/524894.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/529999.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 5fee1ff93e6f38a3c3eff0e26470c619a2e9ea57 onto `release/1.11.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/529999)

**Pull request merged**